### PR TITLE
fix NullPointerException in refactored code

### PR DESCRIPTION
**Pull request by barrydegraaff**
at com.zimbra.oauth.token.handlers.impl.NextCloudTokenHandler.refreshAccessToken(NextCloudTokenHandler.java:84)

### CODE REVIEW

File: src/java/com/zimbra/oauth/token/handlers/impl/NextCloudTokenHandler.java [hunk 1]
```diff
@@ -70,19 +70,17 @@ public static String refreshAccessToken(Account acct, String client) throws Serv
         final OAuthInfo oauthInfo = new OAuthInfo(new HashMap<String, String>());
         final String refreshToken = OAuth2DataSource.getRefreshToken(dataSource);
 
-        String value = com.zimbra.oauth.token.utilities.LdapOauthHelper.getFirstConfig(Provisioning.A_zimbraOAuthConsumerCredentials,
-                client, acct) ;
-        String indVa [] = value.split(":");
+        String [] values = LdapOauthHelper.loadConfiguration(acct, Provisioning.A_zimbraOAuthConsumerCredentials,
```
Comment (desouzas) [inline src/java/com/zimbra/oauth/token/handlers/impl/NextCloudTokenHandler.java:73]: This won't work. If you use loadConfiguration you'll get all configs - including the other clients.

File: src/java/com/zimbra/oauth/token/handlers/impl/NextCloudTokenHandler.java [hunk 2]
```diff
@@ -70,19 +70,17 @@ public static String refreshAccessToken(Account acct, String client) throws Serv
...
-        String redirecUri = value.substring(0, value.lastIndexOf(":"));
+        values = LdapOauthHelper.loadConfiguration(acct, Provisioning.A_zimbraOAuthConsumerRedirectUri,
+            client) ;
+        String redirecUri = values[0].substring(0, values[0].lastIndexOf(":"));
         final String clientRedirectUri = redirecUri;
-
-        value = LdapOauthHelper.getFirstConfig(Provisioning.A_zimbraOAuthConsumerAPIScope,client, acct);
-        String nextCloudUrl =value.substring(0, value.lastIndexOf(":"));
-
+        values = LdapOauthHelper.loadConfiguration(acct,  Provisioning.A_zimbraOAuthConsumerAPIScope,client);
```
Comment (desouzas) [inline src/java/com/zimbra/oauth/token/handlers/impl/NextCloudTokenHandler.java:82]:
```suggestion
        values = LdapOauthHelper.loadConfiguration(acct,  Provisioning.A_zimbraOAuthConsumerAPIScope, client + "_noop");
```
Revert all of these changes, and just use this instead to fix.

File: src/java/com/zimbra/oauth/token/handlers/impl/NextCloudTokenHandler.java [hunk 3]
```diff
@@ -70,19 +70,17 @@ public static String refreshAccessToken(Account acct, String client) throws Serv
...
         final String clientSecret = indVa[1];
-        value = LdapOauthHelper.getFirstConfig(Provisioning.A_zimbraOAuthConsumerRedirectUri,
-            client, acct) ;
-        String redirecUri = value.substring(0, value.lastIndexOf(":"));
+        values = LdapOauthHelper.loadConfiguration(acct, Provisioning.A_zimbraOAuthConsumerRedirectUri,
+            client) ;
+        String redirecUri = values[0].substring(0, values[0].lastIndexOf(":"));
         final String clientRedirectUri = redirecUri;
-
-        value = LdapOauthHelper.getFirstConfig(Provisioning.A_zimbraOAuthConsumerAPIScope,client, acct);
```
Comment (desouzas) [inline src/java/com/zimbra/oauth/token/handlers/impl/NextCloudTokenHandler.java:83]:
```
value = LdapOauthHelper.getFirstConfig(Provisioning.A_zimbraOAuthConsumerAPIScope, client + "_noop", acct);
```

Revert all of these changes, and just use this instead to fix.

Comment (barrydegraaff) [reply src/java/com/zimbra/oauth/token/handlers/impl/NextCloudTokenHandler.java:83]: rolled back the changes and re-did them per your comment and added a try-catch with an understandable error message for the admin.

**Comment by girishbhamare** [comment]
FYI...
Testing notes: As per Barry's recent changes, I have built new zm-oauth-social.jar and zm-oauth-social-common.jar and tested with nextcloud set up. Its working fine for me.

**Review by jrjena136 — APPROVED** [review]

**Review by dasiyogesh — APPROVED** [review]

**Comment by jrjena136** [comment]
commit 82c05d938d4369e8b7c17752d8994a1b0f51b916 (HEAD -> develop, origin/develop, origin/barrydegraaff-patch-1, origin/HEAD)
Author: Barry de Graaff <info@barrydegraaff.tk>
Date:   Tue Sep 8 18:05:34 2020 +0200

    fix NullPointerException in refactored code

    at com.zimbra.oauth.token.handlers.impl.NextCloudTokenHandler.refreshAccessToken(NextCloudTokenHandler.java:84)

    Update NextCloudTokenHandler.java

**Pull request merged**